### PR TITLE
added ignoring unparseable messages

### PR DESCRIPTION
**Pull request by pompon0**
This fixes the python tests which were broken by https://github.com/near/nearcore/commit/0d879ac
nayduck run: https://nayduck.near.org/#/run/2491
There are 8 tests that are still failing, but they started failing even before my PR was merged.

### CODE REVIEW

File: chain/network/src/peer/peer_actor.rs [hunk 1]
```diff
@@ -792,11 +792,20 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for PeerActor {
...
+                    self.send_message_or_log(&PeerMessage::HandshakeFailure(
+                        self.my_node_info.clone(),
+                        HandshakeFailureReason::ProtocolVersionMismatch {
+                            version: PROTOCOL_VERSION,
+                            oldest_supported_version: PEER_MIN_ALLOWED_PROTOCOL_VERSION,
+                        },
+                    ));
+                    return;
+                    // Connection will be closed by a handshake timeout
+                }
```
Comment (mina86) [inline chain/network/src/peer/peer_actor.rs:808]: This should be a separate commit since this is a fix for another issue than one described in commit message. There may also be some discussion whether this should be checked after parsing a handshake at the same time Borsh is checked.

File: pytest/lib/peer.py [hunk 2]
```diff
@@ -43,9 +43,12 @@ async def recv(self, expected=None):
...
                 return None
-
-            response = BinarySerializer(schema).deserialize(
-                response_raw, PeerMessage)
+            try:
+                response = BinarySerializer(schema).deserialize(
+                    response_raw, PeerMessage)
+            except IndexError:
+                # unparsable message, ignore.
+                continue
```
Comment (mina86) [inline pytest/lib/peer.py:51]: This feels wrong. You’re forcing the tests to use Borsh which means that you’re effectively disabling all those tests for protobufs plus this only postpones the problem since once Borsh gets disabled those tests will start failing again.

Comment (pompon0) [reply pytest/lib/peer.py:51]: I'm not forcing anything, these tests were using borsh and they are allowed to. These tests are poorly written, because they assume the underlying encoding by partially reimplementing the borsh encoding - and that's what "feels wrong". I'm not disabling anything just not providing extra coverage. I provide coverage from rust tests, python tests have only disadvantages.

Comment (mina86) [reply pytest/lib/peer.py:51]: The tests were assuming Borsh because that was the network encoding node was using.  You’re now adding a new encoding without updating the tests to support it.  Since protobufs are going to be what nodes will use to talk with each other, you are effectively disabling testing of what is actually running in production.

Comment (pompon0) [reply pytest/lib/peer.py:51]: there is a difference between increasing amount of coverable logic and decreasing absolute coverage.

Comment (mina86) [reply pytest/lib/peer.py:51]: Are you going to fix the tests once you remove Borsh support then? Why not do it now?

Comment (pompon0) [reply pytest/lib/peer.py:51]: > Are you going to fix the tests once you remove Borsh support then?

Definitely not leaving them broken, but I hoped to provide coverage with rust tests to get rid of this ad-hoc borsh implementation.

> Why not do it now?

Great question
1. Because it's weekend :)
2. Because I assumed you like nayduck green, so I wanted to fix the tests ASAP (without disabling them, because THAT would decrease the coverage)
3. Because this fix proves that borsh fallback works (and that is important).
4. Because I hate python and post-submit tests.

Comment (mina86) [reply pytest/lib/peer.py:51]: OK, so can you file an Issue describing the tests need to be updated to use protobufs and add a TODO here?

By the way, with protobufs the code should be simpler since you can compile proto to Python and use that directly instead of having a hand-written definitions and deserialisers.

> Because this fix proves that borsh fallback works (and that is important).

There is upgradable.py which tests new and old nodes can communicate so that’s not as crucial.

> Because I hate python and post-submit tests.

You’re welcome to rewrite things in Rust.

File: chain/network/src/peer/peer_actor.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/peer/peer_actor.rs]: done

File: pytest/lib/peer.py (hunk 2 above)
Comment (pompon0) [reply pytest/lib/peer.py]: added TODO.

> By the way, with protobufs the code should be simpler since you can compile proto to Python and use that directly instead of having a hand-written definitions and deserialisers.

Yes, partially, the fix [WIP] is here https://github.com/near/nearcore/commits/gprusak-python-protobuf
The drawbacks are:
- I'm not removing borsh completely from communication, so non-network-related part of encoding is still there.
- I have no idea how to automate code generation on proto update, or even just enforce regenerating the python proto code on presubmit.

**Review by mina86 — APPROVED** [review]
